framework/source/class/qx/lang/Type.js

```diff
@@ -182,6 +183,24 @@ qx.Bootstrap.define("qx.lang.Type",
         this.getClass(value) === "Error" ||
         value instanceof Error)
       );
+    },
+
+
+    /**
+     * Whether the value is a Promise.
+     *
+     * @param value {var} Value to check.
+     * @return {Boolean} Whether the value is a Promise.
+     *
+     * @ignore(Promise)
+     */
+isPromise : function(value)
+    {
+      var nat = (value instanceof (Promise || window.Promise));
```

Comment:
I think this will crash if Promise is undefined. Try this in the console to see why I think so:
```
var nat = (value instanceof (xxx || window.Promise))
```